finagle-http/src/main/scala/com/twitter/finagle/http/filter/ModifyContentAndHeaderFieldFilter.scala

```diff
@@ -0,0 +1,55 @@
+package com.twitter.finagle.http.filter
+
+import com.twitter.finagle.http.{Fields, Request, Response, Status}
+import com.twitter.finagle._
+import com.twitter.util.Future
+
+/**
+ * Modify a content and header field filter in a 204 and 304 response.
+ * Remove a message-body and modify a Content-Length header field if necessary.
+ *
+ * @see [[https://tools.ietf.org/html/rfc2616 "Hypertext Transfer Protocol -- HTTP/1.1"]]
+ * @see [[https://tools.ietf.org/html/rfc7230 "Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing"]]
+ * section-3.3.2 on `Content-Length` for more information.
+ */
+private[finagle] class ModifyContentAndHeaderFieldFilter[Req <: Request] extends SimpleFilter[Req, Response] {
+  import ModifyContentAndHeaderFieldFilter._
+
+  def apply(request: Req, service: Service[Req, Response]): Future[Response] = {
+    service(request) map { rep =>
+      if (mustNotIncludeContent(rep.status)) {
+        rep.clearContent()
```

Comment:
Sure. Though I've added a log here in a new commit, (as you suggested) it might be better to just `fail` here instead of modifying and sending a message? So that users can notice that there's something wrong happening?